src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbBuffer.cs

```diff
@@ -661,119 +772,141 @@ internal void ZeroMemory()
 
         internal Guid ReadGuid(int offset)
         {
-            // faster than Marshal.PtrToStructure(offset, typeof(Guid))
-            byte[] buffer = new byte[16];
-            ReadBytes(offset, buffer, 0, 16);
-            return new Guid(buffer);
+            offset += BaseOffset;
+            Debug.Assert(0 == offset % ADP.PtrSize, "invalid alignment");
+            bool mustRelease = false;
+            try
+            {
+                DangerousAddRef(ref mustRelease);
+                IntPtr ptr = ADP.IntPtrOffset(DangerousGetHandle(), offset);
+                unsafe
+                {
+                    return new Guid(new ReadOnlySpan<byte>(ptr.ToPointer(), sizeof(Guid)));
+                }
+            }
+            finally
+            {
+                if (mustRelease)
+                {
+                    DangerousRelease();
+                }
+            }
         }
         internal void WriteGuid(int offset, Guid value)
         {
-            // faster than Marshal.Copy(value.GetByteArray()
-            StructureToPtr(offset, value);
+            ReadOnlySpan<Guid> spanGuid = MemoryMarshal.CreateReadOnlySpan(ref Unsafe.AsRef(value), 1);
+            WriteBytes(offset, MemoryMarshal.AsBytes(spanGuid));
```

Comment:
Actually, with an `unsafe` method, we could just write:

```suggestion
            WriteBytes(offset, new ReadOnlySpan<byte>(&value, sizeof(Guid)));
```

Reply:
Or better still, eliminate the overhead of `ReadOnlySpan<T>`:

```suggestion
            offset += BaseOffset;
            Validate(offset, sizeof(Guid));
            Debug.Assert(0 == offset % 4, "invalid alignment");

            bool mustRelease = false;

            try
            {
                DangerousAddRef(ref mustRelease);

                unsafe
                {
                    byte* addr = (byte*)DangerousGetHandle() + (uint)offset;
                    Unsafe.WriteUnaligned(addr, value);
                }
            }
            finally
            {
                if (mustRelease)
                {
                    DangerousRelease();
                }
            }
```

Reply:
indeed, it's faster (unsafe ftw !)